src/lpython/parser/parser.yy

```diff
@@ -488,13 +488,13 @@ if_statement
     ;
 
 for_statement
-    : KW_FOR tuple_item KW_IN expr ":" sep statements {
+    : KW_FOR id_item KW_IN expr ":" sep statements {
```

Comment:
Thanks for this PR @krshrimali!
As given in the Python documentation: https://docs.python.org/3/reference/compound_stmts.html#grammar-token-python-grammar-for_stmt
The target_list (id_item in this case) can have identifier, "(" [target_list] ")", "[" [target_list] "]", attributeref, subscription, slicing and * target. So, I think we cannot replace tuple_item with id_item.

Reply:
Oh, sorry! I wasn't aware of this, I'll take a look post work today, and get back to you once and if I've anything as a solution to this. :) Thanks!